SetVideoSource

When i try to set a second url in function setViewSource for the second player it says "IllegalStateException".

Code example 
```
VidstaPlayer player = (VidstaPlayer) findViewById(R.id.player);
player.setVideoSource("http://www.quirksmode.org/html5/videos/big_buck_bunny.mp4");
...
...
public void changeUrl(String url){
   player.setVideoSource(url);
}
```

Attached Stack Trace:

E/MediaPlayer: attachNewPlayer called in state 32
E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     Process: com.**, PID: 24376
                                                                     java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=131272, result=-1, data=Intent { dat=file:///storage/emulated/0/Pictures/**/VID_20170811_103648.mp4 flg=0x1 (has extras) }} to activity {com.**/com.**.NamedActivity}: java.lang.IllegalStateException
                                                                         at android.app.ActivityThread.deliverResults(ActivityThread.java:3852)
                                                                         at android.app.ActivityThread.handleSendResult(ActivityThread.java:3895)
                                                                         at android.app.ActivityThread.access$1600(ActivityThread.java:154)
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1445)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                         at android.os.Looper.loop(Looper.java:148)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5582)
                                                                         at java.lang.reflect.Method.invoke(Native Method)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                      Caused by: java.lang.IllegalStateException
                                                                         at android.media.MediaPlayer._setDataSource(Native Method)
                                                                         at android.media.MediaPlayer.setDataSource(MediaPlayer.java:1128)
                                                                         at android.media.MediaPlayer.setDataSource(MediaPlayer.java:1113)
                                                                         at android.media.MediaPlayer.setDataSource(MediaPlayer.java:1092)
                                                                         at android.media.MediaPlayer.setDataSource(MediaPlayer.java:1069)
                                                                         at android.media.MediaPlayer.setDataSource(MediaPlayer.java:1023)
                                                                         at uk.co.jakelee.vidsta.VidstaPlayer.setUpVideoPlayer(VidstaPlayer.java:376)
                                                                         at uk.co.jakelee.vidsta.VidstaPlayer.setVideoSource(VidstaPlayer.java:362)
                                                                         at uk.co.jakelee.vidsta.VidstaPlayer.setVideoSource(VidstaPlayer.java:357)
                                                                         at com.**.fragments.NamedFragment.addVideo(NamedFragment.java:500)
                                                                         at com.**.fragments.NamedFragment.addMedia(NamedFragment.java:485)
                                                                         at com.**.fragments.NamedFragment.onActivityResult(NamedFragment.java:470)
                                                                         at android.support.v4.app.FragmentActivity.onActivityResult(FragmentActivity.java:164)
                                                                         at com.**.NamedActivity.onActivityResult(NamedActivity.java:115)
                                                                         at android.app.Activity.dispatchActivityResult(Activity.java:6526)
                                                                         at android.app.ActivityThread.deliverResults(ActivityThread.java:3848)
                                                                         at android.app.ActivityThread.handleSendResult(ActivityThread.java:3895) 
                                                                         at android.app.ActivityThread.access$1600(ActivityThread.java:154) 
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1445) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                         at android.os.Looper.loop(Looper.java:148) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5582) 
                                                                         at java.lang.reflect.Method.invoke(Native Method) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 

--- a/vidsta/src/main/java/uk/co/jakelee/vidsta/VidstaPlayer.java
+++ b/vidsta/src/main/java/uk/co/jakelee/vidsta/VidstaPlayer.java
@@ -373,6 +373,7 @@ public class VidstaPlayer extends FrameLayout implements TextureView.SurfaceText
         }
         videoPlayer.setSurface(surface);
         try {
+            videoPlayer.reset();
             videoPlayer.setDataSource(getContext(), videoSource, headers);
             videoPlayer.prepareAsync();
         } catch (IOException e) {

